Content.Server/Paper/PaperSystem.cs

```diff
@@ -135,10 +162,10 @@ public bool TryStamp(EntityUid uid, string stampName, string stampState, PaperCo
             if (!paperComp.StampedBy.Contains(Loc.GetString(stampName)))
             {
                 paperComp.StampedBy.Add(Loc.GetString(stampName));
-                if (paperComp.StampState == null && TryComp<AppearanceComponent>(uid, out var appearance))
+                if (paperComp.StampState == null && HasComp<AppearanceComponent>(uid))
```

Comment:
you're doing double component checks when you should really just be trycomping and passing it as a param.